platform/node/index.d.ts

```diff
@@ -0,0 +1,39 @@
+declare module '@maplibre/maplibre-gl-native' {
+  export type RequestResponse = {
+    data: ArrayBuffer;
+    modified?: Date;
+    expires?: Date;
+    etag?: string;
+  };
+
+  export type MapOptions = {
+    request: (
+      request: { url: string; kind: number },
+      callback: (error?: Error, response?: RequestResponse) => void,
+    ) => void;
+    ration?: number;
+  };
+
+  export type RenderOptions = {
+    zoom?: number; // defaults to 0
+    width?: number; // px, defaults to 512
+    height?: number; // px, defaults to 512
+    center?: [number, number]; // coordinates, defaults to [0,0]
+    bearing?: number; // degrees, counter-clockwise from north, defaults to 0
+    pitch?: number; // degrees, arcing towards the horizon, defaults to 0
+    classes?: string[]; // array of strings
+  };
+
+  export class Map {
+    constructor(mapOptions: MapOptions);
+
+    load: (style: any) => void;
+
+    render: (
+      renderOptions: RenderOptions,
+      callback: (...args: [error: Error, buffer: undefined] | [error: undefined, buffer: Uint8Array]) => void,
```

Comment:
I assume it's either error without buffer OR buffer without error – or could at some point both be available at the same time?

Reply:
unfortunately no, I'm not familiar with typescript types. from https://github.com/maptiler/tileserver-gl/blob/master/src/serve_rendered.js#L1202-L1226 it seems like return data (not always an error)